src/registers/model_specific.rs

```diff
@@ -725,4 +753,69 @@ mod x86_64 {
             }
         }
     }
+
+    impl Apic {
+        /// Reads the IA32_APIC_BASE.
+        ///
+        /// The APIC_BASE must be supported on the CPU, otherwise a general protection exception will
+        /// occur. Support can be detected using the `cpuid` instruction.
+        #[inline]
+        pub fn read() -> (PhysFrame, ApicFlags) {
+            let (frame, flags) = Self::read_raw();
+            (frame, ApicFlags::from_bits_truncate(flags))
+        }
+
+        /// Reads the raw IA32_APIC_BASE.
+        ///
+        /// The APIC_BASE must be supported on the CPU, otherwise a general protection exception will
+        /// occur. Support can be detected using the `cpuid` instruction.
+        #[inline]
+        pub fn read_raw() -> (PhysFrame, u64) {
+            let raw = unsafe { Self::MSR.read() };
+            // extract bits 32 - 51 (incl.)
+            let addr = PhysAddr::new(raw & 0x_000F_FFFF_FFFF_F000);
+            let frame = PhysFrame::containing_address(addr);
```

Comment:
```suggestion
            // extract bits 32 - 51 (incl.)
            let addr = PhysAddr::new_truncate(raw);
            let frame = PhysFrame::containing_address(addr);
```
`PhysAddr::new_truncate` throws away the upper bits and `PhysFrame::containing_address` throws away to lower bits.

The comment has a typo "32 -> 51" -> "12 - 51".